### PR TITLE
Make com.webos.service.mediacontroller|bluetooth2 work on LuneOS

### PR DESCRIPTION
**Pull request by Herrie82**
Seems WAM isn't happy without com.webos.mediacontroller.service, even if disabled. 

Unfortunately com.webos.mediacontroller.service had ugly hardcoded values for the OSE targets, trying to work around this by making it more generic.

Also added Install sections in the service files for com.webos.service.bluetooth2|mediacontroller so they get started on boot. The errors in Testr are now gone, however still no audio playback for the audio tests or in the browser via speaker or Bluetooth. Audio works in cardshell.

### CODE REVIEW

File: meta-luneos/recipes-webos/com.webos.service.mediacontroller/com.webos.service.mediacontroller/0001-com.webos.service.mediacontroller-Make-it-work-gener.patch [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+                                    __FUNCTION__, adapterName.c_str(), adapterAddress.c_str());
+-
+-#if defined(PLATFORM_RASPBERRYPI4)
+-      if(("raspberrypi4 #2" == adapterName) || ("raspberrypi4" == adapterName) ||
+-         ("raspberrypi4-64 #2" == adapterName) || ("raspberrypi4-64" == adapterName) ||
+-         ("qemux86-64 #2" == adapterName) || ("qemux86-64" == adapterName)) {
+-#elif defined(PLATFORM_SA8155)
+-      if ("sa8155 Bluetooth hci0" == adapterName) {
+-#endif
++      if (("" != adapterName) && ("sa8155 Bluetooth hci1" != adapterName)) {
```
Comment (Herrie82) [inline meta-luneos/recipes-webos/com.webos.service.mediacontroller/com.webos.service.mediacontroller/0001-com.webos.service.mediacontroller-Make-it-work-gener.patch:50]: @Tofee Not 100% sure about this "solution" here. Happy for suggestions for something else.

Comment (Tofee) [reply meta-luneos/recipes-webos/com.webos.service.mediacontroller/com.webos.service.mediacontroller/0001-com.webos.service.mediacontroller-Make-it-work-gener.patch:50]: To stay a bit closer to the original version, you could write something like this:
```c
#if defined(PLATFORM_GENERIC)
      if (!adapterName.empty()) {
#elif defined(PLATFORM_SA8155)
      if ("sa8155 Bluetooth hci0" == adapterName) {
#endif
```

File: meta-luneos/recipes-webos/com.webos.service.mediacontroller/com.webos.service.mediacontroller/0001-com.webos.service.mediacontroller-Make-it-work-gener.patch [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+         deviceSetId = "RSE-L";
+         std::string payload = "{\"adapterAddress\":\"" + adapterAddress + "\",\"subscribe\":true}";
+         PMLOG_INFO(CONST_MODULE_MCS,"%s : payload = %s for first adapter",__FUNCTION__, payload.c_str());
+@@ -170,7 +163,7 @@ bool MediaControlService::onBTAdapterQueryCb(LSHandle *lshandle, LSMessage *mess
+             PMLOG_ERROR(CONST_MODULE_MCS,"%s : LSCall failed for BT device/getStatus", __FUNCTION__);
+         }
+       }
+-      else if ("sa8155 Bluetooth hci1" == adapterName) {
++      else if (("" != adapterName) && ("sa8155 Bluetooth hci1" == adapterName)) {
```
Comment (Tofee) [inline meta-luneos/recipes-webos/com.webos.service.mediacontroller/com.webos.service.mediacontroller/0001-com.webos.service.mediacontroller-Make-it-work-gener.patch:59]: unneeded here. If it's "sa8155 Bluetooth hci1", then it's not empty

File: meta-luneos/recipes-webos/com.webos.service.mediacontroller/com.webos.service.mediacontroller/0001-com.webos.service.mediacontroller-Make-it-work-gener.patch (hunk 1 above)
Comment (Herrie82) [reply meta-luneos/recipes-webos/com.webos.service.mediacontroller/com.webos.service.mediacontroller/0001-com.webos.service.mediacontroller-Make-it-work-gener.patch:50]: I'm more worried if it does what it's intend to do. I don't really understand what they were trying to do ;) 

On pinephonepro the adaptername shows up as "pinephonepro" which I guess is what we expect
`Dec 22 10:07:53 pinephonepro com.webos.service.mediacontroller[1624]: [] [pmlog] mediacontroller MediaControlService {} onBTAdapterQueryCb : adapterName : pinephonepro, adapterAddress : aa:aa:aa:aa:aa:aa
Dec 22 10:07:53 pinephonepro com.webos.service.mediacontroller[1624]: [] [pmlog] mediacontroller MediaControlService {} onBTAdapterQueryCb : payload = {"adapterAddress":"aa:aa:aa:aa:aa:aa","subscribe":true} for first adapter
Dec 22 10:07:53 pinephonepro com.webos.service.mediacontroller[1624]: [] [pmlog] mediacontroller MediaControlService {} onBTAdapterQueryCb : adapterAddress = aa:aa:aa:aa:aa:aa deviceSetId = RSE-L [182]
Dec 22 10:07:53 pinephonepro com.webos.service.mediacontroller[1624]: [] [pmlog] mediacontroller MediaControlPrivate {}  setBTAdapterInfo
`

File: meta-luneos/recipes-webos/com.webos.service.mediacontroller/com.webos.service.mediacontroller/0001-com.webos.service.mediacontroller-Make-it-work-gener.patch (hunk 2 above)
Comment (Herrie82) [reply meta-luneos/recipes-webos/com.webos.service.mediacontroller/com.webos.service.mediacontroller/0001-com.webos.service.mediacontroller-Make-it-work-gener.patch:59]: Ah true, I'll change that

File: meta-luneos/recipes-webos/com.webos.service.mediacontroller/com.webos.service.mediacontroller/0001-com.webos.service.mediacontroller-Make-it-work-gener.patch (hunk 1 above)
Comment (Tofee) [reply meta-luneos/recipes-webos/com.webos.service.mediacontroller/com.webos.service.mediacontroller/0001-com.webos.service.mediacontroller-Make-it-work-gener.patch:50]: I guess they just want to make sure we go into that "if" only for the adapter we build for. They do like to have per-target binaries, though I don't really see the big advantage of doing that here.

**Comment by Herrie82** [comment]
@tofee Seems to work OK on PPP, qemux86-64 and hammerhead-halium

**Pull request merged**